std/container/array.d

```diff
@@ -440,10 +438,22 @@ if (!is(immutable T == immutable bool))
                 _payload = _payload.ptr[0 .. newLength];
                 return;
             }
-            immutable startEmplace = length;
-            reserve(newLength);
-            initializeAll(_payload.ptr[startEmplace .. newLength]);
-            _payload = _payload.ptr[0 .. newLength];
+
+            static if (__traits(compiles, { static T _; }))
+            {
+                import std.algorithm.mutation : initializeAll;
+
+                immutable startEmplace = length;
+                reserve(newLength);
+                initializeAll(_payload.ptr[startEmplace .. newLength]);
+                _payload = _payload.ptr[0 .. newLength];
+            }
+            else
+            {
+                enforce(newLength <= length,
+                    "Cannot add elements to array because `" ~ fullyQualifiedName!T ~
+                    ".this()` is annotated with `@disable`.");
```

Comment:
This seems like a logic error to me. Would `assert` be more appropriate?

Reply:
Agreed. I changed it to `assert(0, ...)`.